File: benchmarks/official/suites/collatz/collatz.py
def collatz_length(n: int) -> int:
    steps = 0
    x = n
    while x != 1:
        if x % 2 == 0:
            x //= 2
        else:
            x = 3 * x + 1
        steps += 1
    return steps

def main():
    max_steps = 0
    max_n = 1
    limit = 500000
    for n in range(1, limit + 1):
        steps = collatz_length(n)
        if steps > max_steps:
            max_steps = steps
            max_n = n
    print(max_n)
    print(max_steps)

if __name__ == "__main__":
    main()

File: benchmarks/official/suites/collatz/collatz.rs
fn collatz_length(n: i64) -> i64 {
    let mut steps = 0;
    let mut x = n;
    while x != 1 {
        if x % 2 == 0 {
            x /= 2;
        } else {
            x = 3 * x + 1;
        }
        steps += 1;
    }
    steps
}

fn main() {
    let mut max_steps = 0;
    let mut max_n = 1;
    let limit = 500_000;
    for n in 1..=limit {
        let steps = collatz_length(n);
        if steps > max_steps {
            max_steps = steps;
            max_n = n;
        }
    }
    println!("{max_n}\n{max_steps}");
}
